components/sup/src/main.rs

```diff
@@ -1460,6 +1460,13 @@ fn valid_topology(val: String) -> result::Result<(), String> {
     }
 }
 
+fn valid_listen(val: String) -> result::Result<(), String> {
+    match GossipListenAddr::from_str(&val) {
+        Ok(_) => Ok(()),
+        Err(_) => Err(format!("Listen address should include both IP and port, eg: '0.0.0.0:9700'"))
+    }
+}
```

Comment:
Though both things that you're validating are IP/port addresses, in the codebase they are represented by distinct types. The `LISTEN_GOSSIP` options are `GossipListenAddr`s, while the `LISTEN_HTTP` options are `http_gateway::ListenAddr`s.

For clarity, they should both have their own validators, which use the appropriate types (even though they are pretty much going to be the same functions).

Reply:
Oh, yes, didn't know. Will do when I'll be on my computer. Thanks!